### ISSUE TITLE
GameGuru MAX - Request - Instances

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#3746]
I would like to request that Instances be added to GameGuru MAX. Instances are copies of the original object that are tied to the original in such a way that if you change the original, the instances all change, too.

For example, create a point light and set it all up (range, color, etc.) then create instances of that point light, such as going down a corridor, etc. Then, later, if you decide to adjust the light (i.e. you think the lighting is too bright or too dim), you only have to adjust the original (first) light and all instances of that light automatically change to match it.

If you bring an object into you scene and create instances of it, then you could make changes to the original (such as changing material settings) and all instances would get the changes as well.

If you add in a particle and then make several instances of it (say you have sparks used at several severed cables), then you'd only need to make adjustments to the first and all the instances would update.

This would save a lot of time over having to make changes to each and every light in a scene, etc.

**Comment by Necrym59** [comment]
Yes i agree - if an existing entity is copied then an option to have a "cloned instance" (link) should be there to have its attribute/characteristics shared by those copies so all can be effected by any change to any one of the cloned instances.

**Comment by fearlesswee** [comment]
Could this be folded into the existing grouping system? Like when you group a bunch of stuff together into a "smart object" you have the option to set it as an "instanced smart object" which will then know that when you make changes to one "smart object" it will update all other instances of that "smart object" in the level?
